### ISSUE TITLE
FMT001: Detect explicit numbered lists instead of AsciiDoc dot-syntax

### ISSUE THREAD

**Issue by raifdmueller** [docToolchain/asciidoc-linter#27]
## Problem

LLMs often generate explicitly numbered lists like:
```asciidoc
1. First item
2. Second item
3. Third item
```

Instead of the proper AsciiDoc ordered list syntax:
```asciidoc
. First item
. Second item
. Third item
```

The explicit numbering is problematic because:
- It doesn't auto-renumber when items are added/removed
- It's not semantic AsciiDoc
- It creates maintenance burden

## Proposed Solution

Add a new rule `FMT001` that detects lines starting with `\d+\. ` (number, dot, space) and suggests using `. ` instead.

## Examples

**Should trigger warning:**
```asciidoc
1. First step
2. Second step
```

**Correct format:**
```asciidoc
. First step
. Second step
```

## Acceptance Criteria
- [ ] New rule `FMT001` in `format_rules.py`
- [ ] Detects `1. `, `2. `, etc. at start of line
- [ ] Suggests using `. ` syntax
- [ ] Tests cover various cases
